frontend/components/MostViewed.tsx

```diff
@@ -147,49 +148,199 @@ const headlineLink = css`
     font-weight: 500;
 `;
 
+const tabsContainer = css`
+    border-bottom: 1px solid ${palette.neutral[86]};
+
+    &::after {
+        content: '';
+        display: block;
+        clear: left;
+
+        ${tablet} {
+            display: none;
+        }
+    }
+
+    ${tablet} {
+        border-bottom: 0;
+    }
+`;
+
+const listTab = css`
+    width: 50%;
+    float: left;
+    border-top: 3px solid ${palette.neutral[93]};
+    background-color: ${palette.neutral[93]};
+
+    ${phablet} {
+        width: 230px;
+    }
+`;
+
+const selectedListTab = css`
+    background-color: ${palette.neutral[100]};
+`;
+
+const tabButton = css`
+    margin: 0;
+    border: 0;
+    background: transparent;
+    padding-left: 10px;
+    padding-right: 6px;
+    padding-top: 4px;
+    text-align: left;
+    text-decoration: none;
+    line-height: 20px;
+    font-size: 14px;
+    font-family: ${serif.headline};
+    font-weight: 600;
+    min-height: 36px;
+    display: block;
+    width: 100%;
+
+    ${tablet} {
+        font-size: 16px;
+    }
+`;
+
 interface Trail {
     url: string;
     linkText: string;
 }
-const fetchTrails: () => Promise<Trail[]> = () =>
-    new Promise((resolve, reject) => {
-        fetch('https://api.nextgen.guardianapps.co.uk/most-read-geo.json?guui')
-            .then(response => {
-                if (!response.ok) {
+
+interface Tab {
+    heading: string;
+    trails: Trail[];
+}
+
+interface Props {
+    sectionName: string;
+}
+
+export class MostViewed extends Component<Props, { selectedTabIndex: number }> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            selectedTabIndex: 0,
+        };
+    }
+
+    public tabSelected(index: number) {
+        this.setState({
+            selectedTabIndex: index,
+        });
+    }
+
+    public render() {
+        return (
+            <div className={container}>
+                <h2 className={heading}>Most Viewed</h2>
+                <AsyncClientComponent f={this.fetchTrails}>
```

Comment:
Is `this` bound correctly?

Reply:
As described in this [post](https://github.com/Microsoft/TypeScript/wiki/'this'-in-TypeScript#use-instance-functions) from Microsoft's typescript wiki you can correctly keep the context of `this` by using an 'instance arrow function" to define a class member, as I've done here with `fetchTrails`.